R/make_split_fun.R

```diff
@@ -124,9 +134,9 @@ make_split_result <- function(values, datasplit, labels, extras = NULL) {
 #' @rdname make_split_result
 #' @param splres list. A list representing the result of splitting.
 #' @export
-add_to_split_result <- function(splres, values, datasplit, labels, extras = NULL) {
+add_to_split_result <- function(splres, values, datasplit, labels, extras = NULL, sub_expr = NULL) {
   validate_split_result(splres)
-  newstuff <- make_split_result(values, datasplit, labels, extras)
+  newstuff <- make_split_result(values, datasplit, labels, extras, subset_exprs = list(sub_expr))
```

Comment:
why `sub_expr` is not defaulting to the `splres` values if sub_expr is NULL?

Reply:
So there are two cases that can happen here:

1. values are strings (this is more common as this is being called in custom code), in which case value_expr(values[.]) will always be NULL, so that wouldn't change anything. 
2. values are already SplitValue objects, in which case they (should) already have their expressions set (if they are not the default, which will still work in most cases), so they don't need to be changed.

Currently, as written, `subset_expr` ends up being ignored when values are already SplitValue objects.

This is defensible behavior, I think (as the construction time of the SplitValue is where the custom expression "should" be set), but probably not optimal. I'll change things slightly to make it easier to override subset expressions if (for some reason) you're reusing the same value object but want different subsetting behavior (which would be weird but I can think of a a corner case you might want to). As written originally you'd have been expected to reset the expressions on the value objects before passing to make_split_result